src/codeGenerator/evaluator.ts

```diff
@@ -66,6 +69,10 @@ export class FunctionClosure {
 
 
 export class Evaluator implements ExpressionVisitor {
+    // Counters used to generate Minecraft function and advancement names
+    fnCounter = 0;
+    advCounter = 0;
```

Comment:
Should these be static fields?

Reply:
I don't think so - assuming there's exactly one evaluator call per datapack input, these IDs should be reset to 0 on each fresh instance.